Saule/Queries/Lambda.cs

```diff
@@ -74,17 +78,42 @@ internal static object TryConvert(string value, Type type)
 
         private static Type GetPropertyType(Type type, string property)
         {
-            var returnType = type.GetProperty(property)?.PropertyType;
-            if (returnType == null)
+            List<string> properties = property.Split('.').ToList();
+            var returnType = type;
+            foreach (string prop in properties)
             {
-                throw new ArgumentException(
-                    $"Property {property} does not exist.",
-                    nameof(property));
+                returnType = returnType.GetProperty(prop)?.PropertyType;
+                if (returnType == null)
+                {
+                    throw new ArgumentException(
+                        $"Property {property} does not exist.",
+                        nameof(property));
+                }
             }
 
             return returnType;
         }
 
+        private static PropertyInfo GetPropertyInfo(Type type, string property)
+        {
+            List<string> properties = property.Split('.').ToList();
+            var tmpType = type;
+            PropertyInfo propertyInfo = tmpType.GetProperty(property);
+            foreach (string prop in properties)
+            {
+                propertyInfo = tmpType.GetProperty(prop);
+                tmpType = tmpType.GetProperty(prop)?.PropertyType;
+                if (tmpType == null)
+                {
+                    throw new ArgumentException(
```

Comment:
Tbh I am not sure. I just copied that over from how it was before, because it's basically the same now as before it just has to do the check in a foreach loop. I can check tho if you want me to.

Reply:
If it's the same it was probably for some good reason.. I can investigate sometime but for now we should leave it.